roles.js

```diff
@@ -100,98 +38,132 @@ exports.init = (c) => {
+    client.getUserRunCount = sql.prepare("SELECT count(*) FROM runs WHERE username = ?;");
+    client.getGameCategories = sql.prepare("SELECT DISTINCT category_id, il FROM runs WHERE game_id = ?;");
+    client.getCategories = sql.prepare("SELECT DISTINCT category_id, game_id, il FROM runs;");
+    client.getUserGames = sql.prepare("SELECT DISTINCT game_id FROM runs WHERE username = ?;");
+    client.getUserWRCount = sql.prepare("SELECT count(*) FROM runs WHERE wr = 1 AND username = ?;");
+    client.addRun = sql.prepare("INSERT INTO runs (game_id, category_id, username, il, wr, placeholder) VALUES (@game_id, @category_id, @username, @il, @wr, @placeholder);");
+    client.deleteRun = sql.prepare("DELETE FROM runs WHERE category_id = ? AND username = ?;");
+    client.deletePlaceholder = sql.prepare("DELETE FROM runs WHERE category_id = ? AND placeholder = 1;");
+
+    // Setup SQL queries for setting/retrieving user discord data
+    client.getUser = sql.prepare("SELECT * FROM discord WHERE username = ?;");
+    client.getAutoConnectedUsers = sql.prepare("SELECT * FROM discord WHERE auto = 1;");
+    client.addUser = sql.prepare("INSERT OR REPLACE INTO discord (username, id, auto) VALUES (@username, @id, @auto);");
+    client.deleteUser = sql.prepare("DELETE FROM discord WHERE username = ?;");
+
+    autoRefreshTimeout = setTimeout(getUsers, 86400000 * Math.ceil(Date.now() / 86400000) - Date.now() + 1);
 }
 
 exports.roleCmds = (lowerMessage, message) => {
+    // Role commands (available anywhere)
     if (lowerMessage.startsWith("!roles reload categories"))
         reloadCategoriesCmd(message);
     else if (lowerMessage.startsWith("!roles reload leaderboards"))
         reloadLeaderboardCmd(message);
-    else if (lowerMessage.startsWith("!roles reload discordaccount"))
-        reloadDiscordAccountCmd(message);
+    else if (lowerMessage.startsWith("!roles autoconnect"))
+        autoConnectCmd(message);
     else if (lowerMessage.startsWith("!roles reload all"))
         reloadAllCmd(message);
     else if (lowerMessage.startsWith("!roles connect"))
         connectCmd(message);
+    else if (lowerMessage.startsWith("!roles disconnect"))
+        disconnectCmd(message);
 }
 
 // !roles reload categories
 reloadCategoriesCmd = (message) => {
-    getCategories(() => {
+    getCategories(message, recursiveUpdating, () => {
         message.channel.send("Updated category data.");
-        saveData();
     });
 }
 
 // !roles reload leaderboards
 reloadLeaderboardsCmd = (message) => {
-    if (Object.keys(leaderboards).length === 0) {
-        message.channel.send("No category data found, try running `!reload categories` first.");
+    if (client.getRunCount['count(*)'] === 0) {
+        message.channel.send("No category data found, try running `!roles reload categories` first.");
         return;
     }
 
     game = message.content.replace(/^!roles reload leaderboards/i, "").trim();
 
-    if (game.toLowerCase() === "all") {
```

Comment:
Would rather just do this straightforward string equality check than add more regex into the mix, it's more readable.

Reply:
changed it back